linera-indexer/src/operations.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines the operations indexer plugin.
+
+use crate::types::IndexerError;
+use async_graphql::{
+    EmptyMutation, EmptySubscription, Object, OneofObject, OutputType, Schema, SimpleObject,
+};
+use linera_base::{
+    crypto::{BcsHashable, CryptoHash},
+    data_types::BlockHeight,
+    identifiers::ChainId,
+};
+use linera_chain::data_types::HashedValue;
+use linera_execution::Operation;
+use linera_views::{
+    common::Context,
+    map_view::MapView,
+    views::{RootView, View, ViewError},
+};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::info;
+
+#[derive(Deserialize, Serialize, Clone, SimpleObject, Debug)]
+pub struct BlockOperation {
+    block: CryptoHash,
+    index: usize,
+    content: linera_execution::Operation,
+}
+
+impl BcsHashable for BlockOperation {}
+
+#[derive(Deserialize, Serialize, Clone, SimpleObject, Debug)]
+pub struct ChainOperation<H: Sync + Send + OutputType> {
+    operation: BlockOperation,
+    height: BlockHeight,
+    previous_operation: Option<CryptoHash>,
+    operation_index: u64,
+    hash: H,
```

Comment:
Is this generic just to deduplicate the hash if it's in the map with the hash as the key?
Then instead of `ChainOperation<bool>` we should use `ChainOperation<()>`, should we? Because the bool is always `false`?
Or maybe just have two (non-generic) types `ChainOperation` and `HashedChainOperation`?

Reply:
it is just that `OutputType` is not implemented for `()` and it was a hassle to implement it...

Reply:
I see; then probably two types (one without the hash, and a wrapper type with the hash) is the best solution?